packages/webviz-core/src/dataProviders/MemoryCacheDataProvider.js

```diff
@@ -106,6 +94,23 @@ export function getBlocksToKeep({
   return { blockIndexesToKeep, newRecentRanges: recentBlockRanges };
 }
 
+// Get the best place to start prefetching a block, given the uncached ranges and the cursor position.
+// In order of preference, we would like to prefetch:
+// - The leftmost uncached block to the right of the cursor, or
+// - The leftmost uncached block to the left of the cursor, if one does not exist to the right.
+//
+// Exported for tests.
+export function getPrefetchStartPoint(uncachedRanges: Range[], cursorPosition: Number): Number {
```

Comment:
```suggestion
export function getPrefetchStartPoint(uncachedRanges: Range[], cursorPosition: number): number {
```

Flow/JS sadly has both `number` and `Number` but we typically use the lowercase one which explicitly denotes the primitive (e.g. `(5 instanceof Number) === false` 😭).

Reply:
Done. Fixed another couple of flow problems, let me know if the fixes aren't great.

Also: I rebased and force-pushed those fixes, but I guess I shouldn't have, right? Should I be waiting until some time later to make the history pretty?

Reply:
I think that's okay. Usually when we merge we squash it all into one commit. And it was very useful for me to read through it commit-by-commit!

Reply:
> Usually when we merge we squash it all into one commit

Anyone reading the git-blame to understand why the code is the way it is will have even less context than you do now. If individual commits add clarity to reviewers, I think it's extra important they be preserved for maintainers.

Reply:
Yeah, I tend to agree with you, but in practice you can look up the original PR in Github (which is linked through the PR number) and get the context that way. It is unfortunate though that the context is not captured in git itself though. If you feel strongly maybe we can discuss your proposal at our next retrospective meeting?